### PR TITLE
Add support to Datacom (Teracom Telematica S/A) device families

### PR DESCRIPTION
**Pull request by lhwolfarth**
- Enhancements:
datacom (DmSwitch)
- Basic support:
datacom-dm1200e
datacom-dm1205e
datacom-dm2500
datacom-dmos

Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by lhwolfarth** [comment]
> Please add test data so we can ensure your change is not broken in the future. Read the docs to find out how: https://docs.librenms.org/Developing/os/Test-Units


https://github.com/librenms/librenms/pull/14259/commits/9147fd93dde4fa157d3610093c85d2a6edb2ca2b

**Comment by murrant** [comment]
Why have you created so many OS?  Wouldn't it be better to fill in the hardware field instead?

File: includes/definitions/discovery/datacom-dm1205e.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+mib: SNMPv2-MIB
+modules:
+    os:
+     hardware: 
+             - SNMPv2-MIB::sysDescr.0
```
Comment (murrant) [inline includes/definitions/discovery/datacom-dm1205e.yaml:5]: https://docs.librenms.org/Developing/os/Initial-Detection/#os-discovery
Use sysDescr_regex instead, what you have done here is call snmp to the device again, when you already have sysDescr

**Comment by lhwolfarth** [comment]
> Why have you created so many OS? Wouldn't it be better to fill in the hardware field instead?

Not shure if there is other way

**Comment by lhwolfarth** [comment]
> Why have you created so many OS? Wouldn't it be better to fill in the hardware field instead?

Not sure if there is other way to implement because each .yaml in fact refer to different OS despite they are from the same company. Each OS have it's own snmp tree structure so please confirm if in this case could be feasible use the hardware field anyway.

**Comment by murrant** [comment]
They don't look that different from this standpoint, but I don't have time to look in depth.  And often when vendors use Broadcom silicon, they leave the fastpath mibs, but implement theirs as well.  (And sometimes later remove the broadcom mibs).

Sorry, but I need to make sure, as fixing os level mistakes is a pain.

File: includes/definitions/discovery/datacom-dm1205e.yaml (hunk 1 above)
Comment (lhwolfarth) [reply includes/definitions/discovery/datacom-dm1205e.yaml]: Just a question. How can I use the information fetched by the sysDescr in hardware field without call the snmp again? Sometimes I don't have the hardware description in another OID then I use the sysDescr.

**Pull request closed without merging**